File: pi-platform-agent/scripts/invoke-cursor-agent.mjs
#!/usr/bin/env node
/**
 * One-shot local Cursor agent for scheduled home-agent runs.
 * Usage: invoke-cursor-agent.mjs <prompt-file> [cwd]
 *
 * Requires CURSOR_API_KEY (typically from ~/.env, sourced by home-agent-run.sh).
 * Install deps: npm install --prefix <pi-platform-agent-dir>
 */
import { readFileSync } from "node:fs";
import { resolve } from "node:path";
import { Agent, CursorAgentError } from "@cursor/sdk";

const promptPath = process.argv[2];
const cwd = resolve(process.argv[3] || process.cwd());

if (!promptPath) {
  console.error("Usage: invoke-cursor-agent.mjs <prompt-file> [cwd]");
  process.exit(2);
}

const apiKey = process.env.CURSOR_API_KEY;
if (!apiKey) {
  console.error("CURSOR_API_KEY is not set");
  process.exit(1);
}

const modelId = process.env.CURSOR_AGENT_MODEL || "composer-2.5";
const prompt = readFileSync(promptPath, "utf8");

console.log(`invoke-cursor-agent: model=${modelId} cwd=${cwd}`);

try {
  const result = await Agent.prompt(prompt, {
    apiKey,
    model: { id: modelId },
    local: { cwd },
  });

  console.log(
    JSON.stringify(
      { status: result.status, id: result.id, hasResult: Boolean(result.result) },
      null,
      2,
    ),
  );
  if (result.result) {
    console.log("--- agent result ---");
    console.log(result.result);
  }
  process.exit(result.status === "error" ? 2 : 0);
} catch (err) {
  if (err instanceof CursorAgentError) {
    console.error(`CursorAgentError: ${err.message} retryable=${err.isRetryable}`);
    process.exit(1);
  }
  throw err;
}

File: pi-platform-agent/scripts/invoke-cursor-agent.py
#!/usr/bin/env python3
"""One-shot local Cursor agent (Python SDK). See invoke-cursor-agent.mjs for Node variant."""
from __future__ import annotations

import os
import sys
from pathlib import Path


def main() -> int:
    if len(sys.argv) < 2:
        print("Usage: invoke-cursor-agent.py <prompt-file> [cwd]", file=sys.stderr)
        return 2

    prompt_path = Path(sys.argv[1])
    cwd = Path(sys.argv[2]).resolve() if len(sys.argv) > 2 else Path.cwd()

    api_key = os.environ.get("CURSOR_API_KEY")
    if not api_key:
        print("CURSOR_API_KEY is not set", file=sys.stderr)
        return 1

    from cursor_sdk import Agent, AgentOptions, LocalAgentOptions

    model = os.environ.get("CURSOR_AGENT_MODEL", "composer-2.5")
    prompt = prompt_path.read_text(encoding="utf-8")

    print(f"invoke-cursor-agent: model={model} cwd={cwd}")

    result = Agent.prompt(
        prompt,
        AgentOptions(
            api_key=api_key,
            model=model,
            local=LocalAgentOptions(cwd=str(cwd)),
        ),
    )

    print({"status": result.status, "id": result.id, "has_result": bool(result.result)})
    if result.result:
        print("--- agent result ---")
        print(result.result)

    return 2 if result.status == "error" else 0


if __name__ == "__main__":
    raise SystemExit(main())
